dev/tasks/tasks.yml

```diff
@@ -16,36 +16,42 @@
 # under the License.
 
 tasks:
+  # supported keys
+  # - name: arbitrary task name
+  #   branch: defaults to name
+  #   platform: osx|linux|win
+  #   template: path of jinja2 templated yml
+  #   params: optional extra parameters
```

Comment:
Having the template be a .travis.yml or .appveyor.yml file only in need of template processing seems potentially inflexible to me. Here is a possible alternative:

* tasks.yml provides the path to a task definition file (which could be YAML)
* The task definition file contains the platform, parameters, and additional information needed to process the task. The idea here would be to make the task definition file more modular; so simple tasks could just be a templated .travis.yml, but we wouldn't be constraining ourselves to that model

My guess is that as time goes on, more of the logic for each task will move from templates to Python code, as a means of improving code reuse.

Reply:
I've already lost in the `YAML forest` :)
I'm trying to update this PR do deploy the artifacts - hopefully until the sync, but these ci-test-loops are insanely time consuming.